File: ts-pr/55/typescript/code-quality-naming/user_manager.ts
class UserManager {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private u: string[] = [];
    private cnt: number = 0;
    private flg: boolean = false;
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    private userList: string[] = [];
    private activeUserCount: number = 0;
    private isSystemInitialized: boolean = false;
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    add(n: string): void {
        this.u.push(n);
        this.cnt++;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    addUser(username: string): void {
        this.userList.push(username);
        this.activeUserCount++;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    del(idx: number): boolean {
        if (idx >= 0 && idx < this.u.length) {
            this.u.splice(idx, 1);
            this.cnt--;
            return true;
        }
        return false;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    removeUserByIndex(userIndex: number): boolean {
        if (userIndex >= 0 && userIndex < this.userList.length) {
            this.userList.splice(userIndex, 1);
            this.activeUserCount--;
            return true;
        }
        return false;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    get cnt_val(): number {
        return this.cnt;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    getUserCount(): number {
        return this.activeUserCount;
    }
    // {/fact}
}

const manager = new UserManager();
manager.add('john');
console.log(`User count: ${manager.cnt_val}`);

File: go-pr/48/go/code-quality-naming/user_manager.go
package main

import "fmt"

type UserManager struct {
	// {fact rule=code-quality-naming@v1.0 defects=1}
	u   []string
	cnt int
	flg bool
	// {/fact}
	
	// {fact rule=code-quality-naming@v1.0 defects=0}
	userList            []string
	activeUserCount     int
	isSystemInitialized bool
	// {/fact}
}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (um *UserManager) add(n string) {
	um.u = append(um.u, n)
	um.cnt++
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (um *UserManager) AddUser(username string) {
	um.userList = append(um.userList, username)
	um.activeUserCount++
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (um *UserManager) del(idx int) bool {
	if idx >= 0 && idx < len(um.u) {
		um.u = append(um.u[:idx], um.u[idx+1:]...)
		um.cnt--
		return true
	}
	return false
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (um *UserManager) RemoveUserByIndex(userIndex int) bool {
	if userIndex >= 0 && userIndex < len(um.userList) {
		um.userList = append(um.userList[:userIndex], um.userList[userIndex+1:]...)
		um.activeUserCount--
		return true
	}
	return false
}
// {/fact}

func main() {
	manager := &UserManager{}
	manager.add("john")
	fmt.Printf("User count: %d\n", manager.cnt)
}